kfs-core/src/main/java/edu/arizona/kfs/gl/batch/BudgetAdjustmentMergeSortStep.java

```diff
@@ -0,0 +1,140 @@
+		
+		// names of the two .done files that we are checking to see if they exist
+		String budgetOfficeDoneFileName = GeneralLedgerConstants.BatchFileSystem.ENTERPRISE_FEED + GeneralLedgerConstants.BatchFileSystem.DONE_FILE_EXTENSION;
+		String laborLedgerDoneFileName = GeneralLedgerConstants.BatchFileSystem.LABOR_GL_ENTRY_FILE + GeneralLedgerConstants.BatchFileSystem.DONE_FILE_EXTENSION;
+		
+		// pull out a list of the .done files in the batch file directory
+		File directory = new File(batchFileDirectoryName);
+		FileFilter doneFileFilter = new SuffixFileFilter(GeneralLedgerConstants.BatchFileSystem.DONE_FILE_EXTENSION);
+		File[] doneFiles = directory.listFiles(doneFileFilter);
+		
+		boolean laborLedgerDoneFileExists = false;
+		boolean budgetOfficeDoneFileExists = false;
+		
+		// check each file in the list to see if 
+		// the files we need to merge exist in the directory
+		for (File doneFile : doneFiles) {
+			String fileName = doneFile.getName();
+			
+			if (fileName.equals(laborLedgerDoneFileName)) {
+				laborLedgerDoneFileExists = true;
+			} 
+			
+			if (fileName.equals(budgetOfficeDoneFileName)){
+				budgetOfficeDoneFileExists = true;
+			}
+		}
+		
+		// checks to see which files actually exists in the directory
+		// then pulls the necessary .data files to be processed and put into the output file.
+		if (laborLedgerDoneFileExists && budgetOfficeDoneFileExists) {	
+			String mergedFile = batchFileDirectoryName + File.separator + GeneralLedgerConstants.BatchFileSystem.BUDGET_ADJUSTMENT_MERGE_FILE + GeneralLedgerConstants.BatchFileSystem.EXTENSION;
+			
+			try {
+				mergeFiles(laborLedgerFileName, budgetOfficeFileName, mergedFile);
+			} catch (Exception e) {
+				LOG.error( "Exception merging the files", e );
+				e.printStackTrace();
+			}
+			
+			BatchSortUtil.sortTextFileWithFields(mergedFile, outputFile, new ScrubberSortComparator());
+			
+		} else if (laborLedgerDoneFileExists && !budgetOfficeDoneFileExists) {
+			// if only the labor ledger .done file exists, process the Labor Ledger File	
+			BatchSortUtil.sortTextFileWithFields(laborLedgerFileName, outputFile, new ScrubberSortComparator());		
+			
+		} else if (budgetOfficeDoneFileExists && !laborLedgerDoneFileExists) {
+			// if only the budget office .done file exists, process the Budget Office File
+			BatchSortUtil.sortTextFileWithFields(budgetOfficeFileName, outputFile, new ScrubberSortComparator());
+			
+		} else {
+			// if neither .done file exists, produce an empty file then process the file
+			String emptyFileName = batchFileDirectoryName + File.separator + "temp" + GeneralLedgerConstants.BatchFileSystem.EXTENSION;
+			
+			// creates the new temp file
+			File emptyFile = new File(emptyFileName);
+			try {
+				emptyFile.createNewFile();
+			} catch (IOException e) {
+				LOG.error("Error creating the temp file", e);
```

Comment:
Exception is caught and logged but batch job allowed to continue even if the file can't be created, no exception is thrown further the rest of the batch job is not aware something went wrong...
I think we need to return false and stop the job at this point since we can't create a new file.